doc/classes/RayCast2D.xml

```diff
@@ -74,7 +74,7 @@
 		<method name="get_collision_point" qualifiers="const">
 			<return type="Vector2" />
 			<description>
-				Returns the collision point at which the ray intersects the closest object.
+				Returns the collision point at which the ray intersects the closest object. If [member hit_from_inside] is [code]true[/code] and the ray starts inside of a collision shape, this function will return the origin point of the ray.
```

Comment:
Are we certain that the *exact* return value is the origin point of the ray? People may be trying to perform exact comparison to check if the ray was inside something, and we need to make sure this assertion is correct.

If this is not the case, I suggest rewording it to mention that the position will be similar, but not necessarily identical to the ray origin.

Reply:
Every time I tried it, it was always the exact same point, even with floating point positions. I'll try to make a test scene in a little bit to do some further testing.

Reply:
Made a script to test this out. It's always EXACTLY the same point. You can test for yourself if you'd like. Just add this to an empty Node2D in an empty scene:

```
extends Node2D

var sample_size: int = 20
@onready var static_body: StaticBody2D = StaticBody2D.new()
@onready var collision_shape: CollisionShape2D = CollisionShape2D.new()
@onready var raycast: RayCast2D = RayCast2D.new()

func _ready():
	collision_shape.shape = RectangleShape2D.new()
	collision_shape.shape.size = Vector2(PI * 10, PI * 10)
	raycast.enabled = true
	raycast.hit_from_inside = true
	raycast.collide_with_bodies = true
	raycast.collide_with_bodies = true
	add_child(static_body)
	add_child(raycast)
	static_body.call_deferred("add_child", collision_shape)
	static_body.call_deferred("set", "global_position", Vector2(collision_shape.shape.size.x / 2, collision_shape.shape.size.y / 2))
	var timer = get_tree().create_timer(1)
	timer.connect("timeout", Callable(self, "call_timer"))

func start_test():
	for i in range(0, sample_size):
		randomize()
		raycast.position = Vector2(randf_range(0, collision_shape.shape.size.x), randf_range(0, collision_shape.shape.size.y))
		randomize()
		raycast.target_position = Vector2(randf_range(-100, 100), randf_range(-100, 100))
		raycast.force_raycast_update()
		if raycast.is_colliding():
			var collision_point = raycast.get_collision_point()
			print("=================================================")
			print("ATTEMPT %s: \n RAY POSITION: %s (RAY TARGET: %s) \n COLLISION POINT: %s \n TEST_SUCCESS: %s" % [i, raycast.global_position, raycast.target_position, collision_point, bool(collision_point == raycast.global_position)])
			print("=================================================")
		else:
			print("(Attempt %s) ERROR: Raycast is not colliding!" % i)

func call_timer():
	start_test()

```